### PR TITLE
feat: Add a consent screen as part of the onboarding process

### PR DESCRIPTION
**Pull request by aymanenadi**
This PR adds logic to render a consent screen if the project has a consent form and if the user did not accept it yet. Declining the consent logs the user out.

https://github.com/lifeomic/react-native-sdk/assets/19395435/1023220a-9ff5-46b5-ac89-50a30e994612

### CODE REVIEW

File: example/package.json [hunk 1]
```diff
@@ -33,6 +33,7 @@
     "react-native-draggable-flatlist": "^3.0.6",
     "react-native-gesture-handler": "^2.9.0",
     "react-native-keychain": "^8.1.1",
+    "react-native-markdown-display": "^7.0.0-alpha.2",
```
Comment (sternetj) [inline example/package.json:36]: Should we use 6.x instead of an alpha version?

File: src/hooks/useConsent.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      !loadingDirectives &&
+      directivesData &&
+      directivesData.items.length > 0 &&
+      directivesData.items[0].status === 'active';
+
+    const hasConsent = !loadingDefaultConsents && !!defaultConsentData;
+
+    return {
+      isLoading: loadingDirectives || loadingDefaultConsents,
+      shouldRenderConsentScreen: hasConsent && !didAcceptConsent,
```
Comment (sternetj) [inline src/hooks/useConsent.ts:99]: Should this be
```suggestion
      shouldRenderConsentScreen: hasConsent || !didAcceptConsent,
```

File: src/hooks/useConsent.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  const useShouldRenderConsentScreen = () => {
+    const { data: directivesData, isLoading: loadingDirectives } =
+      useConsentDirectives();
+    const { data: defaultConsentData, isLoading: loadingDefaultConsents } =
+      useDefaultConsent();
+
+    const didAcceptConsent =
+      !loadingDirectives &&
+      directivesData &&
+      directivesData.items.length > 0 &&
```
Comment (markdlv) [inline src/hooks/useConsent.ts:92]: Somewhat nit but also nice for defensiveness on `items` - might simplify 91-92 to something like `!!directivesData?.items?.length`

File: src/screens/ConsentScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+          text: t('logout', 'Logout'),
+          style: 'destructive',
+          onPress: async () => {
+            await updateConsentDirective(false);
+            await logout({});
+          },
+        },
+      ],
+    );
+  };
```
Comment (markdlv) [inline src/screens/ConsentScreen.tsx:51]: Might want to make all three of these methods above utilize `useCallback`

**Review by markdlv — APPROVED** [review]
Approving in case all of my & Teddy's comments end up being answered and/or handled in subsequent PRs

File: src/hooks/useConsent.ts (hunk 2 above)
Comment (aymanenadi) [reply src/hooks/useConsent.ts:99]: No, I should've named `hasConsent` better, but it should be something like `defaultConsentExists`. We only want to render the consent screen if the project has a consent, and the user did not accept it. I'll rename the variable to be more descriptive.

**Pull request merged**